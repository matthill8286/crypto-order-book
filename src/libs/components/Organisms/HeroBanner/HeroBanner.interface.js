
//# sourceMappingURL=HeroBanner.interface.js.map