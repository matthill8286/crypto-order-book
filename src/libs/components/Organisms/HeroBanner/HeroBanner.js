import React, { useRef } from 'react';
import { useHistory } from 'react-router-dom';
import { Card } from "../../Atoms/Card";
import { Picture } from "../../Atoms/Picture";
import { media, styled } from "../../../styles";
var StyledHeroImageWrapper = styled.div.withConfig({
  displayName: "HeroBanner__StyledHeroImageWrapper",
  componentId: "sc-10ip6rx-0"
})(["cursor:", ";width:100%;display:flex;height:", ";", "{height:", ";}", "{height:", ";}"], function (_ref) {
  var isLinking = _ref.isLinking;
  return isLinking ? 'pointer' : 'default';
}, function (_ref2) {
  var height = _ref2.height;
  return height && height.mobile && "".concat(height.mobile, "px");
}, media.md, function (_ref3) {
  var height = _ref3.height;
  return height && height.tablet && "".concat(height.tablet, "px");
}, media.lg, function (_ref4) {
  var height = _ref4.height;
  return height && height.desktop && "".concat(height.desktop, "px");
});
export var HeroBanner = function HeroBanner(_ref5) {
  var badges = _ref5.badges,
      link = _ref5.link,
      badgeActionType = _ref5.badgeActionType,
      height = _ref5.height,
      color = _ref5.color,
      heroImages = _ref5.heroImages,
      _ref5$elevation = _ref5.elevation,
      elevation = _ref5$elevation === void 0 ? 0 : _ref5$elevation,
      _ref5$elevationHover = _ref5.elevationHover,
      elevationHover = _ref5$elevationHover === void 0 ? 4 : _ref5$elevationHover,
      children = _ref5.children,
      onClick = _ref5.onClick;
  var history = useHistory();

  var onTileClick = function onTileClick(event) {
    if (typeof onClick === 'function') {
      onClick(event);
    }

    if (link !== null && link !== void 0 && link.to) return history.push(link.to);
    if (link !== null && link !== void 0 && link.href) return window.location.href = link.href;
  };

  var ref = useRef(null);
  var addElevationHover = link !== null && link !== void 0 && link.to || link !== null && link !== void 0 && link.href ? elevationHover : 0;
  return React.createElement(StyledHeroImageWrapper, {
    ref: ref,
    onClick: onTileClick,
    height: height,
    isLinking: !!(link !== null && link !== void 0 && link.to) || !!(link !== null && link !== void 0 && link.href)
  }, React.createElement(Card, {
    elevation: elevation,
    elevationHover: addElevationHover,
    shape: "rounded-small",
    surface: color,
    display: "flex",
    noBorder: 'none',
    fullHeight: true,
    contentWidth: "100%",
    badges: badges,
    badgeActionType: badgeActionType
  }, heroImages && React.createElement(Picture, {
    src: heroImages.xl,
    srcLg: heroImages.lg,
    srcMd: heroImages.md,
    srcSm: heroImages.sm,
    objectFit: "cover",
    height: "100%",
    width: "100%"
  }), children));
};
//# sourceMappingURL=HeroBanner.js.map