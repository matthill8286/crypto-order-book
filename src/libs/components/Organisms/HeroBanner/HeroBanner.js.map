{"version":3,"sources":["../../../../src/components/Organisms/HeroBanner/HeroBanner.tsx"],"names":["React","useRef","useHistory","Card","Picture","media","styled","StyledHeroImageWrapper","div","isLinking","height","mobile","md","tablet","lg","desktop","HeroBanner","badges","link","badgeActionType","color","heroImages","elevation","elevationHover","children","onClick","history","onTileClick","event","to","push","href","window","location","ref","addElevationHover","xl","sm"],"mappings":"AAAA,OAAOA,KAAP,IAAgBC,MAAhB,QAA8B,OAA9B;AACA,SAASC,UAAT,QAA2B,kBAA3B;AACA,SAASC,IAAT;AACA,SAASC,OAAT;AACA,SAASC,KAAT,EAAgBC,MAAhB;AAIA,IAAMC,sBAAsB,GAAGD,MAAM,CAACE,GAAV;AAAA;AAAA;AAAA,6FAChB;AAAA,MAAGC,SAAH,QAAGA,SAAH;AAAA,SAAoBA,SAAS,GAAG,SAAH,GAAe,SAA5C;AAAA,CADgB,EAIhB;AAAA,MAAGC,MAAH,SAAGA,MAAH;AAAA,SAAgBA,MAAM,IAAIA,MAAM,CAACC,MAAjB,cAA8BD,MAAM,CAACC,MAArC,OAAhB;AAAA,CAJgB,EAMxBN,KAAK,CAACO,EANkB,EAOd;AAAA,MAAGF,MAAH,SAAGA,MAAH;AAAA,SAAgBA,MAAM,IAAIA,MAAM,CAACG,MAAjB,cAA8BH,MAAM,CAACG,MAArC,OAAhB;AAAA,CAPc,EAUxBR,KAAK,CAACS,EAVkB,EAWd;AAAA,MAAGJ,MAAH,SAAGA,MAAH;AAAA,SAAgBA,MAAM,IAAIA,MAAM,CAACK,OAAjB,cAA+BL,MAAM,CAACK,OAAtC,OAAhB;AAAA,CAXc,CAA5B;AAeA,OAAO,IAAMC,UAAqC,GAAG,SAAxCA,UAAwC,QAW/C;AAAA,MAVJC,MAUI,SAVJA,MAUI;AAAA,MATJC,IASI,SATJA,IASI;AAAA,MARJC,eAQI,SARJA,eAQI;AAAA,MAPJT,MAOI,SAPJA,MAOI;AAAA,MANJU,KAMI,SANJA,KAMI;AAAA,MALJC,UAKI,SALJA,UAKI;AAAA,8BAJJC,SAII;AAAA,MAJJA,SAII,gCAJQ,CAIR;AAAA,mCAHJC,cAGI;AAAA,MAHJA,cAGI,qCAHa,CAGb;AAAA,MAFJC,QAEI,SAFJA,QAEI;AAAA,MADJC,OACI,SADJA,OACI;AACJ,MAAMC,OAAO,GAAGxB,UAAU,EAA1B;;AAEA,MAAMyB,WAAW,GAAG,SAAdA,WAAc,CAACC,KAAD,EAA6B;AAC/C,QAAI,OAAOH,OAAP,KAAmB,UAAvB,EAAmC;AACjCA,MAAAA,OAAO,CAACG,KAAD,CAAP;AACD;;AAGD,QAAIV,IAAJ,aAAIA,IAAJ,eAAIA,IAAI,CAAEW,EAAV,EAAc,OAAOH,OAAO,CAACI,IAAR,CAAaZ,IAAI,CAACW,EAAlB,CAAP;AACd,QAAIX,IAAJ,aAAIA,IAAJ,eAAIA,IAAI,CAAEa,IAAV,EAAgB,OAAQC,MAAM,CAACC,QAAP,CAAgBF,IAAhB,GAAuBb,IAAI,CAACa,IAApC;AACjB,GARD;;AAUA,MAAMG,GAAG,GAAGjC,MAAM,CAAC,IAAD,CAAlB;AACA,MAAMkC,iBAAiB,GAAGjB,IAAI,SAAJ,IAAAA,IAAI,WAAJ,IAAAA,IAAI,CAAEW,EAAN,IAAYX,IAAZ,aAAYA,IAAZ,eAAYA,IAAI,CAAEa,IAAlB,GAAyBR,cAAzB,GAA0C,CAApE;AAEA,SACE,oBAAC,sBAAD;AACE,IAAA,GAAG,EAAEW,GADP;AAEE,IAAA,OAAO,EAAEP,WAFX;AAGE,IAAA,MAAM,EAAEjB,MAHV;AAIE,IAAA,SAAS,EAAE,CAAC,EAACQ,IAAD,aAACA,IAAD,eAACA,IAAI,CAAEW,EAAP,CAAD,IAAc,CAAC,EAACX,IAAD,aAACA,IAAD,eAACA,IAAI,CAAEa,IAAP;AAJ5B,KAKE,oBAAC,IAAD;AACE,IAAA,SAAS,EAAET,SADb;AAEE,IAAA,cAAc,EAAEa,iBAFlB;AAGE,IAAA,KAAK,EAAC,eAHR;AAIE,IAAA,OAAO,EAAEf,KAJX;AAKE,IAAA,OAAO,EAAC,MALV;AAME,IAAA,QAAQ,EAAE,MANZ;AAOE,IAAA,UAAU,MAPZ;AAQE,IAAA,YAAY,EAAC,MARf;AASE,IAAA,MAAM,EAAEH,MATV;AAUE,IAAA,eAAe,EAAEE;AAVnB,KAWGE,UAAU,IACT,oBAAC,OAAD;AACE,IAAA,GAAG,EAAEA,UAAU,CAACe,EADlB;AAEE,IAAA,KAAK,EAAEf,UAAU,CAACP,EAFpB;AAGE,IAAA,KAAK,EAAEO,UAAU,CAACT,EAHpB;AAIE,IAAA,KAAK,EAAES,UAAU,CAACgB,EAJpB;AAKE,IAAA,SAAS,EAAC,OALZ;AAME,IAAA,MAAM,EAAC,MANT;AAOE,IAAA,KAAK,EAAC;AAPR,IAZJ,EAsBGb,QAtBH,CALF,CADF;AAgCD,CA3DM","sourcesContent":["import React, { useRef } from 'react'\nimport { useHistory } from 'react-router-dom'\nimport { Card } from '@/components/Atoms/Card'\nimport { Picture } from '@/components/Atoms/Picture'\nimport { media, styled } from '@/styles'\nimport { HeightMap } from '@/types'\nimport { HeroBannerProps } from './HeroBanner.interface'\n\nconst StyledHeroImageWrapper = styled.div<{ height?: HeightMap; isLinking?: boolean }>`\n  cursor: ${({ isLinking }) => (isLinking ? 'pointer' : 'default')};\n  width: 100%;\n  display: flex;\n  height: ${({ height }) => height && height.mobile && `${height.mobile}px`};\n\n  ${media.md} {\n    height: ${({ height }) => height && height.tablet && `${height.tablet}px`};\n  }\n\n  ${media.lg} {\n    height: ${({ height }) => height && height.desktop && `${height.desktop}px`};\n  }\n`\n\nexport const HeroBanner: React.FC<HeroBannerProps> = ({\n  badges,\n  link,\n  badgeActionType,\n  height,\n  color,\n  heroImages,\n  elevation = 0,\n  elevationHover = 4,\n  children,\n  onClick,\n}) => {\n  const history = useHistory()\n\n  const onTileClick = (event: React.MouseEvent) => {\n    if (typeof onClick === 'function') {\n      onClick(event)\n    }\n\n    // navigate to link\n    if (link?.to) return history.push(link.to)\n    if (link?.href) return (window.location.href = link.href)\n  }\n\n  const ref = useRef(null)\n  const addElevationHover = link?.to || link?.href ? elevationHover : 0\n\n  return (\n    <StyledHeroImageWrapper\n      ref={ref}\n      onClick={onTileClick}\n      height={height}\n      isLinking={!!link?.to || !!link?.href}>\n      <Card\n        elevation={elevation}\n        elevationHover={addElevationHover}\n        shape=\"rounded-small\"\n        surface={color}\n        display=\"flex\"\n        noBorder={'none'}\n        fullHeight\n        contentWidth=\"100%\"\n        badges={badges}\n        badgeActionType={badgeActionType}>\n        {heroImages && (\n          <Picture\n            src={heroImages.xl}\n            srcLg={heroImages.lg}\n            srcMd={heroImages.md}\n            srcSm={heroImages.sm}\n            objectFit=\"cover\"\n            height=\"100%\"\n            width=\"100%\"\n          />\n        )}\n        {children}\n      </Card>\n    </StyledHeroImageWrapper>\n  )\n}\n"],"file":"HeroBanner.js"}