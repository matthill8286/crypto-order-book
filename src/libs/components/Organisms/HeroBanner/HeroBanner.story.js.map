{"version":3,"sources":["../../../../src/components/Organisms/HeroBanner/HeroBanner.story.tsx"],"names":["text","storiesOf","React","Heading","StorybookRouterWrapper","HeroBanner","badges","id","name","knobs","link","to","height","mobile","tablet","desktop","heroImages","lg","md","sm","xl","module","add","width","display"],"mappings":";;AAAA,SAASA,IAAT,QAAqB,wBAArB;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAOC,KAAP,MAAkB,OAAlB;AACA,SAASC,OAAT;AACA,SAASC,sBAAT;AACA,SAASC,UAAT;AAEA,IAAMC,MAAM,GAAG,CACb;AAAEC,EAAAA,EAAE,EAAE,CAAN;AAASC,EAAAA,IAAI,EAAE;AAAf,CADa,EAEb;AAAED,EAAAA,EAAE,EAAE,CAAN;AAASC,EAAAA,IAAI,EAAE;AAAf,CAFa,CAAf;;AAKA,IAAMC,KAAK,GAAG,SAARA,KAAQ,GAAM;AAClB,SAAO;AACLC,IAAAA,IAAI,EAAE;AACJC,MAAAA,EAAE,EAAEX,IAAI,CAAC,SAAD,EAAY,GAAZ;AADJ,KADD;AAILY,IAAAA,MAAM,EAAE;AACNC,MAAAA,MAAM,EAAE,GADF;AAENC,MAAAA,MAAM,EAAE,GAFF;AAGNC,MAAAA,OAAO,EAAE;AAHH,KAJH;AASLC,IAAAA,UAAU,EAAE;AACVC,MAAAA,EAAE,EAAE,sFADM;AAEVC,MAAAA,EAAE,EAAE,sFAFM;AAGVC,MAAAA,EAAE,EAAE,sFAHM;AAIVC,MAAAA,EAAE,EAAE;AAJM;AATP,GAAP;AAgBD,CAjBD;;AAmBAnB,SAAS,CAAC,oCAAD,EAAuCoB,MAAvC,CAAT,CACGC,GADH,CACO,SADP,EACkB,YAAM;AACpB,SACE,oBAAC,sBAAD,QACE;AACE,IAAA,KAAK,EAAE;AACLC,MAAAA,KAAK,EAAE,MADF;AAELC,MAAAA,OAAO,EAAE;AAFJ;AADT,KAKE,oBAAC,UAAD;AAAY,IAAA,GAAG,EAAE,eAAef,KAAK,GAAGG;AAAxC,KAAoDH,KAAK,EAAzD,EALF,CADF,CADF;AAWD,CAbH,EAcGa,GAdH,CAcO,sBAdP,EAc+B,YAAM;AACjC,SACE,oBAAC,sBAAD,QACE;AACE,IAAA,KAAK,EAAE;AACLC,MAAAA,KAAK,EAAE,MADF;AAELC,MAAAA,OAAO,EAAE;AAFJ;AADT,KAKE,oBAAC,UAAD;AAAY,IAAA,GAAG,EAAE,eAAef,KAAK,GAAGG;AAAxC,KAAoDH,KAAK,EAAzD;AAA6D,IAAA,MAAM,EAAEH;AAArE,KALF,CADF,CADF;AAWD,CA1BH,EA2BGgB,GA3BH,CA2BO,gDA3BP,EA2ByD,YAAM;AAC3D,SACE,oBAAC,sBAAD,QACE,oBAAC,UAAD,eACMb,KAAK,EADX;AAEE,IAAA,MAAM,EAAE;AAAEI,MAAAA,MAAM,EAAE,GAAV;AAAeC,MAAAA,MAAM,EAAE,GAAvB;AAA4BC,MAAAA,OAAO,EAAE;AAArC,KAFV;AAGE,IAAA,UAAU,EAAE;AACVE,MAAAA,EAAE,EACA,sFAFQ;AAGVC,MAAAA,EAAE,EACA,sFAJQ;AAKVC,MAAAA,EAAE,EACA,sFANQ;AAOVC,MAAAA,EAAE,EACA;AARQ;AAHd,MAaE,oBAAC,OAAD;AAAS,IAAA,KAAK,EAAC,SAAf;AAAyB,IAAA,OAAO,EAAC,IAAjC;AAAsC,IAAA,SAAS,EAAE,QAAjD;AAA2D,IAAA,MAAM,EAAE;AAAnE,2CAbF,CADF,CADF;AAqBD,CAjDH","sourcesContent":["import { text } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport React from 'react'\nimport { Heading } from '@/components/Atoms/Typography'\nimport { StorybookRouterWrapper } from '@/utils/StorybookWrapper'\nimport { HeroBanner } from './HeroBanner'\n\nconst badges = [\n  { id: 0, name: 'New' },\n  { id: 1, name: '100% Awesome!' },\n]\n\nconst knobs = () => {\n  return {\n    link: {\n      to: text('Link To', '#'),\n    },\n    height: {\n      mobile: 350,\n      tablet: 300,\n      desktop: 400,\n    },\n    heroImages: {\n      lg: 'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm',\n      md: 'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm',\n      sm: 'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm',\n      xl: 'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm',\n    },\n  }\n}\n\nstoriesOf('Design System/Organisms/HeroBanner', module)\n  .add('Default', () => {\n    return (\n      <StorybookRouterWrapper>\n        <div\n          style={{\n            width: '100%',\n            display: 'flex',\n          }}>\n          <HeroBanner key={'ImageTile_' + knobs().height} {...knobs()} />\n        </div>\n      </StorybookRouterWrapper>\n    )\n  })\n  .add('With promoted badged', () => {\n    return (\n      <StorybookRouterWrapper>\n        <div\n          style={{\n            width: '100%',\n            display: 'flex',\n          }}>\n          <HeroBanner key={'ImageTile_' + knobs().height} {...knobs()} badges={badges} />\n        </div>\n      </StorybookRouterWrapper>\n    )\n  })\n  .add('Slim version With Feature Hero Image & Heading', () => {\n    return (\n      <StorybookRouterWrapper>\n        <HeroBanner\n          {...knobs()}\n          height={{ mobile: 100, tablet: 125, desktop: 150 }}\n          heroImages={{\n            lg:\n              'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm',\n            md:\n              'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm',\n            sm:\n              'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm',\n            xl:\n              'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm',\n          }}>\n          <Heading scale=\"level-3\" padding=\"md\" textAlign={'center'} margin={'auto'}>\n            What we can offer our professionals\n          </Heading>\n        </HeroBanner>\n      </StorybookRouterWrapper>\n    )\n  })\n"],"file":"HeroBanner.story.js"}