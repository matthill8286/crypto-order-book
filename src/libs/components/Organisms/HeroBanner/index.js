export * from "./HeroBanner";
//# sourceMappingURL=index.js.map