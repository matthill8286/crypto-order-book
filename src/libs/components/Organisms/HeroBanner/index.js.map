{"version":3,"sources":["../../../../src/components/Organisms/HeroBanner/index.ts"],"names":[],"mappings":"AAAA","sourcesContent":["export * from './HeroBanner'\n"],"file":"index.js"}