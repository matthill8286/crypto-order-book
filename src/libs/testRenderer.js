import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { saiyanTheme } from "./styles/sc-vars-saiyan";
export function renderWithTheme(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, tree)).toJSON();
}
export function renderWithThemeAndRouter(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Router, null, tree))).toJSON();
}
export function renderWithRouter(tree) {
  return renderer.create(React.createElement(Router, null, tree)).toJSON();
}
export function mountWithTheme(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, tree));
}
export function mountWithThemeAndRouter(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, React.createElement(Router, null, tree)));
}
//# sourceMappingURL=testRenderer.js.map