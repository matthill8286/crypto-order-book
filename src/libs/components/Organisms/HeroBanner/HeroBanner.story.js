function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import { Heading } from "../../Atoms/Typography";
import { StorybookRouterWrapper } from "../../../utils/StorybookWrapper";
import { HeroBanner } from "./HeroBanner";
var badges = [{
  id: 0,
  name: 'New'
}, {
  id: 1,
  name: '100% Awesome!'
}];

var knobs = function knobs() {
  return {
    link: {
      to: text('Link To', '#')
    },
    height: {
      mobile: 350,
      tablet: 300,
      desktop: 400
    },
    heroImages: {
      lg: 'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm',
      md: 'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm',
      sm: 'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm',
      xl: 'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm'
    }
  };
};

storiesOf('Design System/Organisms/HeroBanner', module).add('Default', function () {
  return React.createElement(StorybookRouterWrapper, null, React.createElement("div", {
    style: {
      width: '100%',
      display: 'flex'
    }
  }, React.createElement(HeroBanner, _extends({
    key: 'ImageTile_' + knobs().height
  }, knobs()))));
}).add('With promoted badged', function () {
  return React.createElement(StorybookRouterWrapper, null, React.createElement("div", {
    style: {
      width: '100%',
      display: 'flex'
    }
  }, React.createElement(HeroBanner, _extends({
    key: 'ImageTile_' + knobs().height
  }, knobs(), {
    badges: badges
  }))));
}).add('Slim version With Feature Hero Image & Heading', function () {
  return React.createElement(StorybookRouterWrapper, null, React.createElement(HeroBanner, _extends({}, knobs(), {
    height: {
      mobile: 100,
      tablet: 125,
      desktop: 150
    },
    heroImages: {
      lg: 'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm',
      md: 'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm',
      sm: 'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm',
      xl: 'https://media.graphcms.com/resize=fit:crop,height:300,width:800/tbT0Znk2RJmb3d6RtCLm'
    }
  }), React.createElement(Heading, {
    scale: "level-3",
    padding: "md",
    textAlign: 'center',
    margin: 'auto'
  }, "What we can offer our professionals")));
});
//# sourceMappingURL=HeroBanner.story.js.map