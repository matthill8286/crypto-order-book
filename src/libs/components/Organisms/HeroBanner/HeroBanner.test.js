import React from 'react';
import { mountWithThemeAndRouter, renderWithThemeAndRouter } from "../../../testRenderer";
import { HeroBanner } from "./HeroBanner";
var images = {
  sm: 'public/images/teasers/500x560.jpg',
  md: 'public/images/teasers/700x560.jpg',
  lg: 'public/images/teasers/1000x360.jpg',
  xl: 'public/images/teasers/1200x360.jpg'
};
describe('HeroBanner', function () {
  it('renders correctly: default teaser with badges', function () {
    var tree = renderWithThemeAndRouter(React.createElement(HeroBanner, {
      color: "primary",
      link: {
        to: '#'
      },
      height: {
        desktop: 400,
        mobile: 350,
        tablet: 375
      },
      badges: [{
        id: 0,
        name: 'New'
      }, {
        id: 1,
        name: '100% Free!'
      }],
      badgeActionType: "prominent",
      heroImages: images
    }));
    expect(tree).toMatchSnapshot();
  });
  it('triggers click handler', function () {
    var clickHandler = jest.fn();
    var tree = mountWithThemeAndRouter(React.createElement(HeroBanner, {
      color: "primary",
      height: {
        desktop: 400,
        mobile: 350,
        tablet: 375
      },
      link: {
        to: '#'
      },
      heroImages: images,
      onClick: clickHandler
    }));
    tree.simulate('click');
    expect(clickHandler).toHaveBeenCalled();
  });
});
//# sourceMappingURL=HeroBanner.test.js.map