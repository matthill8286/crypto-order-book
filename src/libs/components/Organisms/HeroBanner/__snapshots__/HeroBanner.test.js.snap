// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroBanner renders correctly: default teaser with badges 1`] = `
<div
  className="HeroBanner__StyledHeroImageWrapper-sc-10ip6rx-0 bxfUcs"
  height={
    Object {
      "desktop": 400,
      "mobile": 350,
      "tablet": 375,
    }
  }
  onClick={[Function]}
>
  <div
    className="Cardstyled__StyledCardWrapper-sc-1gxmhl7-6 gyZXAf"
  >
    <div
      className="Cardstyled__StyledBadgeLineWrapper-sc-1gxmhl7-0 cFLGQx"
    >
      <div
        className="Badgestyled__StyledBadgeLineWrapper-sc-1xo0r7x-1 ejODHn"
      >
        <div
          className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
        >
          <span
            className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 ePrcMg"
            fontSize="xxs"
          >
            <div
              className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
            >
              New
            </div>
          </span>
        </div>
        <div
          className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
        >
          <span
            className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 ePrcMg"
            fontSize="xxs"
          >
            <div
              className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
            >
              100% Free!
            </div>
          </span>
        </div>
      </div>
    </div>
    <div
      className="Cardstyled__StyledCard-sc-1gxmhl7-1 iKphfo"
      display="flex"
      elevation={0}
      shape="rounded-small"
    >
      <div
        className="Picturestyled__StyledPicture-sc-18jsup7-0 jIsvqb"
        height="100%"
        width="100%"
      >
        <picture
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <source
            media="(max-width: 511px)"
            srcSet="public/images/teasers/500x560.jpg"
          />
          <source
            media="(max-width: 751px)"
            srcSet="public/images/teasers/700x560.jpg"
          />
          <source
            media="(max-width: 1007px)"
            srcSet="public/images/teasers/1000x360.jpg"
          />
          <img
            alt="image"
            height="100%"
            loading="eager"
            src="public/images/teasers/1200x360.jpg"
            width="100%"
          />
        </picture>
      </div>
    </div>
  </div>
</div>
`;
